packages/flutter_tools/lib/src/web/file_generators/flutter_js.dart

```diff
@@ -40,6 +157,15 @@ _flutter.loader = null;
       // Resolver for the pending promise returned by loadEntrypoint.
       this._didCreateEngineInitializerResolve = null;
 
+      // TODO: Make FlutterLoader extend EventTarget once Safari is mature enough
+      // to support EventTarget() constructor.
+      // @see: https://caniuse.com/mdn-api_eventtarget_eventtarget
+      this._eventTarget = document.createElement("custom-event-target");
+
+      // The event of the synthetic CustomEvent that we use to signal that the
+      // entrypoint is loaded.
+      this._eventName = "flutter:entrypoint-loaded";
+
       // Called by Flutter web.
       // Bound to `this` now, so "this" is preserved across JS <-> Flutter jumps.
       this.didCreateEngineInitializer = this._didCreateEngineInitializer.bind(this);
```

Comment:
You can avoid this by making `_didCreateEngineInitializer` an arrow function and then you don't have to explicitly bind. It looks like we're already using arrow functions in this file.

Reply:
Let me try this again. I think I had to do this because of how Dart JS-interop grabs the method (as a tear-off of the object) rather than calling it on the object itself. (This might be a problem of my JS-interop definition, too, but that one might require changes in the engine repo).

Experimenting.

Reply:
This is what I get when I do it "properly" (just removing the `_` from `_didCreateEngineInitializer`):

```
TypeError: Cannot read properties of undefined (reading '_didCreateEngineInitializerResolve')
flutter2.js 208:23                                                  didCreateEngineInitializer
lib/ui/initialization.dart 97:12                                    webOnlyWarmupEngine
```

`this` is undefined on the JS-interop call from Dart, [here](https://github.com/flutter/engine/blob/main/lib/web_ui/lib/initialization.dart#L97). I think the problem is how I defined the callback in dart: [here](https://github.com/flutter/engine/blob/main/lib/web_ui/lib/src/engine/js_interop/js_loader.dart#L19-L20).

(I think a more proper solution is to wrap the `loader` API in some JS dummy class, and then use the `didCreateEngineInitializer`, but I'm not 100% sure.)

Reply:
Don't worry about then, go ahead and leave it as-is. Certainly not worth any additional headaches just to avoid one `bind` call